ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PushQueryPublisher.java

```diff
@@ -135,30 +66,10 @@ public synchronized void subscribe(final Flow.Subscriber<Collection<StreamedRow>
     log.info("Running query {}", queryMetadata.getQueryId().toString());
     queryMetadata.start();
 
-    subscriber.onSubscribe(subscription);
-  }
-
-
-  private static void recordMetrics(
-          final ScalablePushQueryMetrics metrics, final ScalablePushQueryMetadata metadata,
-          final long startTimeNanos) {
-
-    final QuerySourceType sourceType = metadata.getSourceType();
-    final RoutingNodeType routingNodeType = metadata.getRoutingNodeType();
-    // Note: we are not recording response size in this case because it is not
-    // accessible in the websocket endpoint.
-    metrics.recordConnectionDuration(startTimeNanos, sourceType, routingNodeType);
-    metrics.recordRowsReturned(metadata.getTotalRowsReturned(),
-        sourceType, routingNodeType);
-    metrics.recordRowsProcessed(metadata.getTotalRowsProcessed(),
-        sourceType, routingNodeType);
-  }
+    final WebSocketSubscriber<StreamedRow> webSocketSubscriber =
+        (WebSocketSubscriber<StreamedRow>) subscriber;
```

Comment:
first thought... Woah, this whole method is really only used for the websocket?

second pass: oh, I guess the whole "Flow.Pub/Sub" ecosystem is only for websocket, iirc.

Maybe in another pass, we can just remove those interfaces and then hoist this type into the parameters for this method, since it seems like the Pub/Sub interfaces are actively impeding our ability to make this code more maintainable.